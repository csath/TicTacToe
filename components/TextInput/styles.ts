import { StyleSheet } from "react-native";
import Colors from "../../constants/Colors";

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.background,
    width: "70%",
    borderBottomWidth: StyleSheet.hairlineWidth,
    paddingVertical: 16,
    borderBottomColor: Colors.borderColor,
  },
  title: {
    color: Colors.text,
    fontSize: 15,
    paddingVertical: 12,
  },
  textInput: {
    fontSize: 18,
    color: Colors.text,
  },
});

export default styles;
