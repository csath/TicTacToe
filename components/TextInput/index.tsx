import React from "react";
import { TextInput as RNTextInput, Text, View } from "react-native";
import Colors from "../../constants/Colors";
import styles from "./styles";

type CustomTextInputProps = {
  title: string;
  titleStyle?: Text["props"]["style"];
} & RNTextInput["props"];

const TextInput = ({
  title,
  titleStyle,
  style,
  ...props
}: CustomTextInputProps) => {
  return (
    <View style={styles.container}>
      <Text style={[styles.title, titleStyle]}>{title}</Text>
      <RNTextInput
        style={[styles.textInput, style]}
        placeholderTextColor={Colors.textDim}
        {...props}
      />
    </View>
  );
};

export default TextInput;
