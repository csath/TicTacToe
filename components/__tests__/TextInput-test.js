import * as React from 'react';
import renderer from 'react-test-renderer';

import TextInput from '../TextInput';

it(`TextInput renders correctly`, () => {
  const tree = renderer.create(<TextInput text='test'/>).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(2);
});
