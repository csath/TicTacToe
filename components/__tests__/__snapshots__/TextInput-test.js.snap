// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#282828",
      "borderBottomColor": "#e3e6e8",
      "borderBottomWidth": 0.5,
      "paddingVertical": 16,
      "width": "70%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 15,
          "paddingVertical": 12,
        },
        undefined,
      ]
    }
  />
  <TextInput
    placeholderTextColor="#969696"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 18,
        },
        undefined,
      ]
    }
    text="test"
  />
</View>
`;
